File: solutions_1483485_0/C++/evilncrazy/tongue.cpp
#include <cstdio>
#include <algorithm>

char to[300];

int main() {
   freopen("in.txt","r",stdin);
   freopen("out.txt","w",stdout);
   
   char* cl = "our language is impossible to understand there are twenty six factorial possibilities so it is okay if you want to just give up";
   char* x = "ejp mysljylc kd kxveddknmc re jsicpdrysi rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd de kr kd eoya kw aej tysr re ujdr lkgc jv";

   for(int i = 0; i < strlen(cl); i++) {
      to[x[i]] = cl[i];
   }
   
   to['y'] = 'a';
   to['e'] = 'o';
   to['q'] = 'z';
   to['z'] = 'q';
   
   int n;
   scanf("%d\n", &n);
   char buf[10000];
   for(int i = 0; i < n; i++) {
      gets(buf);
      
      printf("Case #%d: ", i+1);
      for(int j = 0; j < strlen(buf); j++) {
         printf("%c", to[buf[j]]);
      }
      
      printf("\n");
   }
}

File: solutions_1483485_0/Python/Junkbot/tongue.py
s = ["ejp mysljylc kd kxveddknmc re jsicpdrysi","rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd","de kr kd eoya kw aej tysr re ujdr lkgc jv"]
e = ["our language is impossible to understand","there are twenty six factorial possibilities","so it is okay if you want to just give up"]

mapping = ['z' for i in xrange(26)]

mapping[0] = 'y'
mapping[25] = 'q'
mapping[ord('o')-ord('a')] = 'e'
#mapping[24] = 'a'
#mapping[4] = 'o'
#mapping[ord('q')-ord('a')] = 'z'

for i in xrange(3):
    for j in xrange(len(s[i])):
        if 'a' <= s[i][j] and s[i][j] <= 'z':
            mapping[ord(s[i][j])-ord('a')] = e[i][j]

#print sorted(mapping)

N = int(raw_input())
for i in xrange(N):
    k = raw_input().strip()
    r = ""

    for j in xrange(len(k)):
        if 'a' <= k[j] and k[j] <= 'z':
            r += mapping[ord(k[j])-ord('a')]
        else:
            r += k[j]
    print "Case #%d: %s" % (i+1, r)
